### PR TITLE
Remove `job_id` from check signature and add tests

### PR DESCRIPTION
**Pull request by konskov**

### CODE REVIEW

File: sql/updates/reverse-dev.sql [hunk 1]
```diff
@@ -18,3 +20,72 @@ AS '@MODULE_PATHNAME@', 'ts_data_node_detach' LANGUAGE C VOLATILE;
...
+SELECT last_value, is_called FROM _timescaledb_config.bgw_job_id_seq;
+
+ALTER EXTENSION timescaledb DROP TABLE _timescaledb_config.bgw_job;
+ALTER EXTENSION timescaledb DROP SEQUENCE _timescaledb_config.bgw_job_id_seq;
+
+DROP TABLE _timescaledb_config.bgw_job;
+
+CREATE SEQUENCE _timescaledb_config.bgw_job_id_seq MINVALUE 1000;
+SELECT pg_catalog.pg_extension_config_dump('_timescaledb_config.bgw_job_id_seq', '');
+SELECT setval('_timescaledb_config.bgw_job_id_seq', last_value, is_called)
```
Comment (fabriziomello) [inline sql/updates/reverse-dev.sql:45]:
```suggestion
SELECT pg_catalog.setval('_timescaledb_config.bgw_job_id_seq', last_value, is_called)
```

File: src/bgw/job.c [hunk 2]
```diff
@@ -67,15 +71,123 @@ ts_bgw_job_start(BgwJob *job, Oid user_oid)
...
+void
+ts_bgw_job_run_config_check(Oid check, int32 job_id, Jsonb *config)
+{
+	List *args =
+		list_make1(makeConst(JSONBOID, -1, InvalidOid, -1, JsonbPGetDatum(config), false, false));
+	FuncExpr *funcexpr =
+		makeFuncExpr(check, VOIDOID, args, InvalidOid, InvalidOid, COERCE_EXPLICIT_CALL);
+
+	/* Nothing to check if there is no check function provided */
+	if (!OidIsValid(check))
```
Comment (fabriziomello) [inline src/bgw/job.c:119]: Shouldn't this check be before the `makeFuncExpr` ?

**Comment by mkindahl** [comment]
Shouldn't you add this to the job-validation-function branch in PR https://github.com/timescale/timescaledb/pull/4425 rather than add it to main?

**Comment by konskov** [comment]
I added to `job-validation-function` so I'm closing this PR.

**Comment by mkindahl** [comment]
> I added to `job-validation-function` so I'm closing this PR.

You know that you can re-parent a PR so that it could be on top of the feature branch? An alternative would have been to do that and then merge the PR with the feature branch.

**Pull request closed without merging**